src/services/store.ts

```diff
@@ -247,11 +247,25 @@ export function setLineFilterCase(caseSensitive: boolean) {
   localStorage.setItem(`${LINE_FILTER_OPTIONS_LOCALSTORAGE_KEY}.caseSensitive`, storedValue);
 }
 
+export function setLineFilterRegex(regex: boolean) {
+  let storedValue = regex.toString();
+  if (typeof regex === 'boolean' && !regex) {
```

Comment:
Same. This function wil always be called with booleans or the signature must reflect the possible types for `regex`.